pycbc/inference/io/__init__.py

```diff
@@ -185,7 +192,7 @@ def check_integrity(filename):
                 _ = fp[group.format(param)][lastidx]
 
 
-def validate_checkpoint_files(checkpoint_file, backup_file):
+def validate_checkpoint_files(checkpoint_file, backup_file, sampler_name):
```

Comment:
Don't make this dependent on the sampler name. That will be a major pain to keep up to date. Just have it be a keyword argument, like `check_nsamples=True`, then below replace the:
```
if sampler_name in ['dynesty', 'cpnest']:
    pass
else:
```
with:
```
if check_nsamples:
```
By making this a kwarg with default of True, you don't have to change any of the other codes, and just have to pass in `check_nsamples=False` into the call to validate_checkpoint_files for dynesty.

Reply:
Ok. Thanks for the suggestion. Let me do this.